kedro-datasets/kedro_datasets/spark/spark_dataset.py

```diff
@@ -304,7 +322,11 @@ def __init__(  # pylint: disable=too-many-arguments
 
         else:
             path = PurePosixPath(filepath)
-
+            if _deployed_on_databricks() and not _path_has_dbfs_prefix(filepath):
+                logger.error(
+                    "Using SparkDataSet on Databricks without the `/dbfs` prefix in the "
+                    "filepath raises an error. Add this prefix to fix the error."
+                )
```

Comment:
Should this be a `DataSetError` since this is considered an invalid config? What kind of error are we getting now, is it some FileNotFound error due to unexpected path?

Reply:
A DataSetError gets raised on trying to save with versioning enabled, but it is not clear from the message why this is. This message is supposed to inform the user. Do you think we should raise an error with a clearer message instead?

Reply:
Yes, I think if this is an invalid code path and we know that it's going to fail we should raise the error earlier, what's the point of delaying until later?

Reply:
I changed this back to a warning in [d3a0a80](https://github.com/kedro-org/kedro-plugins/pull/114/commits/d3a0a80bccbd7306180edca7211617bb81294eeb) and refer to 'undefined behaviour' instead.

I also modified the doc string to make it clearer to the users that they need to specify this `/dbfs` prefix.

Reply:
> Yes, I think if this is an invalid code path and we know that it's going to fail we should raise the error earlier, what's the point of delaying until later?

Ah sorry, I didn't see this reply before I wrote my last message.

I would agree with you but there is one subtlety, which is that if you specify a version in `SparkDataSet`, file path resolution will work. I wouldn't like to raise an error here as this would be a breaking change for the users who do specify a version. Currently, it is only the automatic resolution of the latest version that is broken.

The alternative would be to add another condition to the `if` statement, checking whether a version has been specified and only raising an error in the case that it wasn't. However, I don't like this, as it leads the user into thinking it is OK to leave out the `/dbfs/` prefix when a version is specified. As far as I understand, they should always be using this prefix. Therefore, I would rather display a warning whenever a `SparkDataSet` is invoked incorrectly in every case.

Reply:
I see, in that case I agree we should not break the existing behaviors and I think the current solution with warning is fine.